specs/commandline/blob.md

```diff
@@ -0,0 +1,381 @@
+
+Usage:
+  notation blob [command]
+
+Available Commands:
+  inspect   inspect a signature associated with a blob
+  sign      produce a detached signature for a given blob
+  verify    verify a signature associated with a blob
+
+Flags:
+  -h, --help   help for blob
+```
+
+### notation blob sign
+
+```text
+Produce a detached signature for a given blob. Signature file will be written to the currently working directory with blob file name + ".sig" + signature format as the file extension. For example, signature file name for "myBlob.bin" will be "myBlob.bin.sig.jws" for JWS signature format or "myBlob.bin.sig.cose" for COSE signature format.
+
+Usage:
+  notation blob sign [flags] <blob_path>
+
+Flags:
+  -d,  --signature-directory string optional path where the blob signature needs to be placed (default: currently working directory) 
+       --media-type string          optional media type of the blob (default: "application/octet-stream")
+  -e,  --expiry duration            optional expiry that provides a "best by use" time for the blob. The duration is specified in minutes(m) and/or hours(h). For example: 12h, 30m, 3h20m
+       --id string                  key id (required if --plugin is set). This is mutually exclusive with the --key flag
+  -k,  --key string                 signing key name, for a key previously added to notation's key list. This is mutually exclusive with the --id and --plugin flags
+       --plugin string              signing plugin name. This is mutually exclusive with the --key flag
+       --plugin-config stringArray  {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values.
+       --signature-format string    signature envelope format, options: "jws", "cose" (default "jws")
+       --force                      skip user confirmation and force overwrite the existing blob signature file 
+  -m,  --user-metadata stringArray  {key}={value} pairs that are added to the signature payload
+  -d,  --debug                      debug mode
+  -v,  --verbose                    verbose mode
+  -h,  --help                       help for sign
+```
+
+### notation blob inspect
+
+```text
+Inspect a signature associated with a blob
+
+Usage:
+  notation blob inspect [flags] <signature_path>
+
+Flags:
+  -o, --output string         output format, options: 'json', 'text' (default "text")
+  -d, --debug                 debug mode
+  -v, --verbose               verbose mode
+  -h, --help                  help for inspect
+```
+
+### notation blob verify
+
+```text
+Verify a signature associated with a blob
+
+Usage:
+  notation blob verify [flags] --signature <signature_path> <blob_path>
+
+Flags:
+  -s, --signature string      location of the blob signature file
+      --media-type string     optional media type of the blob to verify
+      --policy-scope string   optional policy scope to verify against. If not provided, notation verifies against wildcard policy if it exists.
```

Comment:
We need to be more clear here. Because a valid version 1.1 trust policy can have two different trust policy statements,  one of them has scope `[oci:*]`, and the other one has scope `[blob:*]`. For `notation blob verify` command, when `--policy-scope` is not set, only `[blob:*]` will be used right? In other words, if `[blob:*]` is missing, even if the user set up `[oci:*]`, the verification should fail?

Reply:
The proposal has evolved from scopes to separate policy documents. See https://hackmd.io/@-KPyDkW6QfGA-pldFa13pA/ByuHffALa

I will update this PR with the new CX